File: Player/src/main/java/ru/testsimpleapps/coloraudioplayer/managers/player/playlist/IPlaylist.java
package ru.testsimpleapps.coloraudioplayer.managers.player.playlist;

public interface IPlaylist extends Cloneable {

    String TEMP_PLAYLIST = "Temp_playlist";

    long ERROR_CODE = -1L;

    boolean add(Object items);

    boolean delete(long id);

    boolean toFirst();

    boolean toLast();

    boolean goToPosition(long position);

    boolean goToId(long id);

    boolean toNext();

    boolean toPrevious();

    long size();

    long position();

    long getPlaylistId();

    long getTrackId();

    String getTrackPath();

    String getTrackName();

    String getTrackArtist();

    String getTrackTitle();

    String getTrackAlbum();

    long getTotalTime();

    long getTrackDuration();

    long getTrackDateModified();

    long find(long position, String name);

    IPlaylist clone();
}

File: player/src/main/java/ru/testsimpleapps/coloraudioplayer/managers/player/playlist/IPlaylist.kt
package ru.testsimpleapps.coloraudioplayer.managers.player.playlist

interface IPlaylist : Cloneable {

    val playlistId: Long

    val trackId: Long

    val trackPath: String

    val trackName: String

    val trackArtist: String

    val trackTitle: String

    val trackAlbum: String

    val totalTime: Long

    val trackDuration: Long

    val trackDateModified: Long

    fun add(items: Any): Boolean

    fun delete(id: Long): Boolean

    fun toFirst(): Boolean

    fun toLast(): Boolean

    fun goToPosition(position: Long): Boolean

    fun goToId(id: Long): Boolean

    fun toNext(): Boolean

    fun toPrevious(): Boolean

    fun size(): Long

    fun position(): Long

    fun find(position: Long, name: String): Long

    public override fun clone(): IPlaylist

    companion object {

        val TEMP_PLAYLIST = "Temp_playlist"

        val ERROR_CODE = -1L
    }
}
